features/FEATURE_LWIP/lwip-interface/EthernetInterface.cpp

```diff
@@ -24,12 +24,35 @@ EthernetInterface::EthernetInterface()
 {
 }
 
+bool validate_str(const char *str, const int len)
```

Comment:
This all seems a bit clunky. This whole routine could just be

    return str != NULL && strlen(str) < len;

You could also have replaced strncpy with snprintf, which doesn't have the no-zero-termination-if-truncating quirk.